src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/DocumentProcessor.java

```diff
@@ -44,10 +46,19 @@ public void uploadPdfFiles(List<Pdf> pdfs, List<ScannableItem> scannedItems) {
             );
 
         if (filesWithoutUrl.isEmpty()) {
-            scannedItems.forEach(item -> item.setDocumentUrl(response.get(item.getFileName())));
+            scannedItems.forEach(item -> {
+                item.setDocumentUrl(response.get(item.getFileName()));
+                item.setDocumentUuid(extractDocumentUuid(response.get(item.getFileName())));
+            });
             scannableItemRepository.saveAll(scannedItems);
         } else {
             throw new DocumentUrlNotRetrievedException(filesWithoutUrl);
         }
     }
+
+    private String extractDocumentUuid(String documentUrl) {
+        //text after the last '/' in the url
+        Matcher matcher = Pattern.compile("([^/]+)(?=[^/]*/?$)").matcher(documentUrl);
+        return matcher.find() ? matcher.group(0) : null; //TODO: throw exception when documentUuid is non-null field
```

Comment:
throw exception when non null? erm..

Reply:
Not sure what to do if there is no uuid.. it can be discussed later.